dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java

```diff
@@ -681,4 +743,115 @@ public EntityType getEntityType() {
         return EntityType.DEVICE;
     }
 
+    private String getHttpPublishCommand(String baseurl, DeviceCredentials deviceCredentials) {
+        if (deviceCredentials.getCredentialsType() == DeviceCredentialsType.ACCESS_TOKEN) {
+            return String.format("curl -v -X POST %s/api/v1/%s/telemetry --header Content-Type:application/json --data " + PAYLOAD,
+                    baseurl, deviceCredentials.getCredentialsId());
+        }
+        return null;
+    }
+
+    private String getMqttPublishCommand(DeviceCredentials deviceCredentials) {
+        return getMqttPublishCommand("v1/devices/me/telemetry", deviceCredentials, " -m " + PAYLOAD);
+    }
+
+    private String getMqttPublishCommand(String deviceTelemetryTopic, DeviceCredentials deviceCredentials, String payload) {
+        DeviceConnectivityInfo mqttProps = deviceConnectivityConfiguration.getConnectivity().get("mqtt");
+
+        StringBuilder command = new StringBuilder("mosquitto_pub -d -q 1");
+        command.append(" -h ").append(mqttProps.getHost());
```

Comment:
if host is null you should use base url, the same for port. Plz check other protocols!